### PR TITLE
[SYCLomatic] Use rewriter to migrate 4 APIs to fix incorrect migration when API is in macro

### PR DESCRIPTION
**Pull request by zhiweij1**
Covered APIs:
cudaGetErrorString
cudaGetErrorName
cudaGetLastError
cudaPeekAtLastError

Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

File: clang/test/dpct/cuda-get-error-string.cu [hunk 1]
```diff
@@ -121,19 +118,19 @@ const char *test_function() {
...
 //CHECK-NEXT:*/
-//CHECK-NEXT:printf("%s\n", "cudaGetErrorName not supported"/*cudaGetErrorName(0)*/);
+//CHECK-NEXT:printf("%s\n", "cudaGetErrorName not supported"/*cudaGetErrorName(cudaSuccess)*/);
   printf("%s\n", cudaGetErrorName(cudaSuccess));
 
 //CHECK:/*
 //CHECK-NEXT:DPCT1009:{{[0-9]+}}: SYCL uses exceptions to report errors and does not use the error codes. The original code was commented out and a warning string was inserted. You need to rewrite this code.
 //CHECK-NEXT:*/
-//CHECK-NEXT:  return "cudaGetErrorName not supported"/*cudaGetErrorName(0)*/;
+//CHECK-NEXT:  return "cudaGetErrorName not supported"/*cudaGetErrorName(cudaSuccess)*/;
```
Comment (AndyCHHuang) [inline clang/test/dpct/cuda-get-error-string.cu]: "is" not supported

**Review by AndyCHHuang — CHANGES_REQUESTED** [review]

File: clang/test/dpct/cuda-get-error-string.cu (hunk 1 above)
Comment (zhiweij1) [reply clang/test/dpct/cuda-get-error-string.cu]: done

**Review by AndyCHHuang — APPROVED** [review]

File: clang/lib/DPCT/CallExprRewriter.cpp [hunk 2]
```diff
@@ -1916,6 +1916,29 @@ createAssignableFactory(
...
+    std::pair<std::string, std::shared_ptr<CallExprRewriterFactoryBase>>
+        &&Input,
+    std::string &&Prefix, std::string &&Suffix) {
+  return std::pair<std::string, std::shared_ptr<CallExprRewriterFactoryBase>>(
+      std::move(Input.first),
+      std::make_shared<InsertAroundRewriterFactory>(
+          Input.second, std::move(Prefix), std::move(Suffix)));
+}
+
+template <class T>
```
Comment (ziranzha) [inline clang/lib/DPCT/CallExprRewriter.cpp]: This overload can be removed

File: clang/lib/DPCT/CallExprRewriter.cpp [hunk 3]
```diff
@@ -2358,6 +2381,8 @@ class CompareArgType {
 };
 
 #define ASSIGNABLE_FACTORY(x) createAssignableFactory(x 0),
+#define INSERT_AROUND_FACTORY(x, PREFIX, SUFFIX)                               \
+  createInsertAroundFactory(x PREFIX, SUFFIX, 0),
```
Comment (ziranzha) [inline clang/lib/DPCT/CallExprRewriter.cpp]: ', 0' is not needed.

File: clang/lib/DPCT/CallExprRewriter.cpp (hunk 2 above)
Comment (zhiweij1) [reply clang/lib/DPCT/CallExprRewriter.cpp]: done

File: clang/lib/DPCT/CallExprRewriter.cpp (hunk 3 above)
Comment (zhiweij1) [reply clang/lib/DPCT/CallExprRewriter.cpp]: done

**Review by ziranzha — APPROVED** [review]
LGTM

**Pull request merged**